server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -215,10 +276,10 @@ public CollectionStatistics collectionStatistics(String field) throws IOExceptio
     }
 
     public DirectoryReader getDirectoryReader() {
-        return engineSearcher.getDirectoryReader();
-    }
-
-    public Engine.Searcher getEngineSearcher() {
-        return engineSearcher;
+        final IndexReader reader = getIndexReader();
```

Comment:
I think we should do a hard cast here. if this is not a directory reader we are in deep s**t. an assert would be good just to make tests fail.

Reply:
++, https://github.com/elastic/elasticsearch/pull/43645/commits/45baded6ece7ca2c978a84faad0dec2c981c8a5a